packages/flutter_tools/templates/cocoapods/Podfile-objc

```diff
@@ -1,30 +1,45 @@
 # Uncomment this line to define a global platform for your project
 # platform :ios, '9.0'
 
-if ENV['FLUTTER_FRAMEWORK_DIR'] == nil
-  abort('Please set FLUTTER_FRAMEWORK_DIR to the directory containing Flutter.framework')
+def parse_KV_file(file,seperator='=')
+  file_abs_path = File.expand_path(file)
+  puts file_abs_path
+  if !File.exists? file_abs_path
+    puts file_abs_path+"Not exists"
+    return [];
+  end
+  pods_ary = []
+  File.foreach(file_abs_path) { |line|
+      puts line
+      plugin = line.split(pattern=seperator)
+      if plugin.length == 2
+        podname = plugin[0].strip()
+        path = plugin[1].strip()
+        podpath = File.expand_path("#{path}", file_abs_path)
+        pods_ary.push({:name => podname,:path=>podpath});
+      else
+        puts "Invalid plugin specification: #{line}"
+      end
+  }
+  return pods_ary
 end
 
 target 'Runner' do
   # Pods for Runner
 
+  # Pods for Runner
   # Flutter Pods
-  pod 'Flutter', :path => ENV['FLUTTER_FRAMEWORK_DIR']
+  flutter_pods_ary = parse_KV_file(".flutter-framework")
```

Comment:
Reading directly out of Generated.xcconfig would be nice without creating a separate file.

Semantically, it fits the natural hierarchy too by create a single source of hierarchical truth. CocoaPods shouldn't work anyway if Generated.xcconfig doesn't exist (meaning the user never ran flutter build/run before or the build is outdated).